src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintTrafficLightAction.kt

```diff
@@ -51,13 +51,16 @@ class SonarLintTrafficLightAction(private val editor: Editor) : AbstractSonarAct
     }
 
     override fun updatePresentation(e: AnActionEvent, project: Project) {
+        if (project.isDisposed) {
+            return
+        }
         e.getData(CommonDataKeys.VIRTUAL_FILE)?.let { file ->
             val presentation = e.presentation
             val isFocusOnNewCode = getService(project, CleanAsYouCodeService::class.java).shouldFocusOnNewCode(project)
             val relevantFindings = getService(project, FindingsCache::class.java).getFindingsForFile(file).filter { !it.isResolved && (!isFocusOnNewCode || it.isOnNewCode()) }
             val issues = relevantFindings.filterIsInstance<LiveIssue>()
             val hotspots = relevantFindings.filterIsInstance<LiveSecurityHotspot>()
-            val taintVulnerabilitiesCount = TaintVulnerabilitiesCache.getStatus(project)?.count(isFocusOnNewCode) ?: 0
+            val taintVulnerabilitiesCount = getService(project, TaintVulnerabilitiesCache::class.java).status?.count(isFocusOnNewCode) ?: 0
```

Comment:
Unrelated to this PR, but it made me think about whether we should check if the project is disposed inside the `getService(project, class)` method?
There can be errors like 
```
	at com.intellij.serviceContainer.ContainerUtilKt.throwAlreadyDisposedError(containerUtil.kt:45)
	at com.intellij.serviceContainer.ComponentManagerImpl.checkThatCreatingOfLightServiceIsAllowed(ComponentManagerImpl.kt:790)
	at com.intellij.serviceContainer.ComponentManagerImpl.getOrCreateLightService(ComponentManagerImpl.kt:770)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:726)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:670)
	at org.sonarlint.intellij.common.util.SonarLintUtils.getService(SonarLintUtils.java:62)
```

Reply:
This is one thing I wanted to explore, I just created [SLI-1236](https://sonarsource.atlassian.net/browse/SLI-1236) after a community report (and I also experienced it myself). Should we have a generic way to deal with disposed services? I will think about it when working on this ticket